Sources/CacheAPI-smf.php

```diff
@@ -86,34 +86,38 @@ public function getData($key, $ttl = null)
 	public function putData($key, $value, $ttl = null)
 	{
 		$key = $this->prefix . strtr($key, ':/', '-_');
-		$cachedir = $this->cachedir;
+		$file_name = '/data_' . $key . '.php';
+		$file_path = $this->cachedir . $file_name;
 
 		// Work around Zend's opcode caching (PHP 5.5+), they would cache older files for a couple of seconds
 		// causing newer files to take effect a while later.
 		if (function_exists('opcache_invalidate'))
-			opcache_invalidate($cachedir . '/data_' . $key . '.php', true);
+			opcache_invalidate($file_path, true);
 
 		if (function_exists('apc_delete_file'))
-			@apc_delete_file($cachedir . '/data_' . $key . '.php');
+			@apc_delete_file($file_path);
 
 		// Otherwise custom cache?
 		if ($value === null)
-			@unlink($cachedir . '/data_' . $key . '.php');
+			@unlink($file_path);
+
 		else
 		{
 			$cache_data = '<' . '?' . 'php if (!defined(\'SMF\')) die; if (' . (time() + $ttl) . ' < time()) $expired = true; else{$expired = false; $value = \'' . addcslashes($value, "\0" . '\\\'') . '\';}' . '?' . '>';
 
 			// Write out the cache file, check that the cache write was successful; all the data must be written
 			// If it fails due to low diskspace, or other, remove the cache file
-			$fileSize = file_put_contents($cachedir . '/data_' . $key . '.php', $cache_data, LOCK_EX);
+			$fileSize = $this->filePutWithLock($file_name, $cache_data);
+
 			if ($fileSize !== strlen($cache_data))
```

Comment:
Line 112 checks the file size and if it doesn't match issues a unlink. filePutWithLock will return a 0 if it can't reach a lock.

Reply:
Changed the logic to not delete the file if the method returned false.  If returned false it means the file wasn't created on this attempt.